src/vs/editor/contrib/suggest/suggestController.ts

```diff
@@ -223,7 +224,18 @@ export class SuggestController implements IEditorContribution {
 		}));
 		this._toDispose.add(this.model.onDidSuggest(e => {
 			if (!e.shy) {
-				let index = this._memoryService.select(this.editor.getModel()!, this.editor.getPosition()!, e.completionModel.items);
+				let index = -1;
+				// Highest priority first
+				const selectors = [...this._selectors].sort((s1, s2) => s2.priority - s1.priority);
```

Comment:
Maybe keep `_selector` always sorted so that you don't need to create and sort a new array each time suggest happens. I'd say it is favorable to pay the price in `registerSelector`